src/components/Products/Filters/Categories.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import useHttpRequest from '../../../hooks/useHttpRequest';
+import Loading from '../../Main/Loading';
+
+const Categories = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const {
+    state: { error, data, loading }
+  } = useHttpRequest({
+    method: 'GET',
+    url: 'categories/'
+  });
+
+  const query = (name) => {
+    const locationSearch = location.search;
+    const urlParams = new URLSearchParams(locationSearch);
```

Comment:
You can write: return `${!locationSearch ? '?' : '&'}category=${name}` instead of all these ifs